plugins/filter/filters.py

```diff
@@ -321,19 +336,29 @@ def client_properties(self, listener_dict, default_ssl_enabled, bouncy_castle_ke
                 final_dict[config_prefix + 'sasl.login.connect.timeout'] = '15000'
                 final_dict[config_prefix + 'sasl.oauthbearer.token.endpoint.url'] = oauth_token_uri
 
-                if oauth_groups_scope == 'none':
-                    final_dict[config_prefix + 'sasl.jaas.config'] = 'org.apache.kafka.common.security.oauthbearer.OAuthBearerLoginModule required \
-                    clientId=\"' + oauth_client_id + '\" clientSecret=\"' + str(oauth_client_password) + '\";'
+                if (oauth_groups_scope == 'none' and (not idp_self_signed)):
+                    final_dict[config_prefix + 'sasl.jaas.config'] = 'org.apache.kafka.common.security.oauthbearer.OAuthBearerLoginModule required ' + \
+                        'clientId=\"' + oauth_client_id + '\" clientSecret=\"' + str(oauth_client_password) + '\";'
```

Comment:
any specifc reason for converting this oauth_client_password to str. Similarly other variables like oauth_client_id can also be converted to str before concat or all can be kept without type conversion

Reply:
here also I used the format that we currently use in all functions, i.e to convert the password with str but user the userid as it is. Looking at the variable type i think it is just a double check for the passwords and not needed but we have been using it historically to avoid any type failures specially in case of encrypted passwords maybe. just a guess

Reply:
Makes sense. Someday when we cleanup we should remove all this and use f strings to cleanup and make things more uniform. for now lets keep it this way